lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -208,6 +208,10 @@ void swift::eraseUsesOfValue(SILValue V) {
     auto *User = UI->getUser();
     UI++;
 
+    // User could be removed at this point.
```

Comment:
One possibility is to collect all the direct and transitive users of the SILValue and remove them one by one.  But that may leave us with some dead, but not removed instructions.